### PR TITLE
Jetty 12.0.x 9072 move core ee classes

### PR DESCRIPTION
**Pull request by janbartel**
Completes #9072 

Moved `Deployable` into `jetty-core/jetty-server`, moved `ConstraintAware` and `ConstraintMapping` into `jetty-ee10/jetty-ee10-servlet`

### CODE REVIEW

**Comment by gregw** [comment]
@janbartel something is still looking for jetty-ee in the CodeQL?

**Comment by janbartel** [comment]
> @janbartel something is still looking for jetty-ee in the CodeQL?
Yes, just pushed a fix - some poms got missed.

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Deployable.java [hunk 1]
```diff
@@ -11,14 +11,20 @@
...
 import java.util.function.Predicate;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+/**
+ * Deployable
+ *
+ * Communicates to the DeploymentManager that a context can be deployed and should
+ * have its defaults initialized.
+ */
```
Comment (gregw) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Deployable.java]:
```suggestion
/**
 * Interface that can be implemented by ContextHandlers within Environments to allow configuration
 * to be passed from the DeploymentManager without dependencies on the Deployment module itself. 
 */
```

**Review by gregw — APPROVED** [review]
but with a comment suggestion. Also check on the CI failure looking for jetty-ee

**Pull request merged**